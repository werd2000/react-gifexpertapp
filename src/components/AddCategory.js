import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    // Desestructuro el useState
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
        // console.log('handleInputChange llamado');
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log('handleSubmit llamado', inputValue);
        if (inputValue.trim().length > 2) {
            setCategories( cate => [inputValue, ...cate ]);
            // mando un string vacio para limppiar el input
            setInputValue('');
        }
        // console.log(e.target.value);
    }

    return (
        <form onSubmit={ handleSubmit }>
            <input 
                type="text" 
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
    
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};