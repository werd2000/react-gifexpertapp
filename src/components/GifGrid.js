// import React, { useEffect, useState } from 'react'
import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

    const {data:images, loading} = useFetchGifs( category );
    // console.log(loading);
    
    // useEffect( () => {
    //     getGifs( category )
    //         .then( images => setImages( images ))
    // }, [ category ])

    // getGifs();

    return (
        <>
            <h3 className='animate__animated animate__fadeIn'>{ category }</h3>

            { loading && <p className='animate__animated animate__flash'>Loading</p>}
            
            <div className='card-grid'>
                { 
                    images.map( image => 
                        <GifGridItem 
                            key={ image.id }
                            { ...image } />)
                        // <li key={ image.id }>
                        //     { image.title }
                        // </li> )
                }
            </div>
        </>
    )
}
