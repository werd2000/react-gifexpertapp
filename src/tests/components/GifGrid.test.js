import React from "react";

import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {

    const category = 'Goku';

    test('Debe mostrarse correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category= { category } />);
        expect( wrapper).toMatchSnapshot();
        
    });

    test('debe mostrar items cuando se cargan imÃ¡genes useFecthGifs', () => {
        const gifs = [{
            id: 'abc',
            url: 'https://google.com',
            title: 'Titulo'
        },
        {
            id: 'bcd',
            url: 'https://localhost/ima.jpg',
            title: 'Titulo2'

        }];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: true
        });
        const wrapper = shallow(<GifGrid category= { category } />);
        // expect(wrapper).toMatchSnapshot();
        // expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    
      
})
