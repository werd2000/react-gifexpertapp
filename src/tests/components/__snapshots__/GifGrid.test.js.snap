// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
